dlib/image_transforms/interpolation.h

```diff
@@ -2001,6 +2001,50 @@ namespace dlib
         extract_image_chip(img, location, chip, interpolate_bilinear());
     }
 
+// ----------------------------------------------------------------------------------------
+
+    template <
+        typename image_type1,
+        typename image_type2,
+        typename interpolation_type
+    >
+    void insert_image_chip (
+        image_type1& image,
+        const image_type2& chip,
+        const chip_details& location,
+        const interpolation_type& interp
+    )
+    {
+        image_view<image_type1> vimg(image);
+        const_image_view<image_type2> vchip(chip);
+        DLIB_CASSERT(static_cast<unsigned long>(vchip.nr()) == location.rows &&
+                     static_cast<unsigned long>(vchip.nc()) == location.cols,
+                     "The chip and the location do not have the same size.")
+        const auto tf = get_mapping_to_chip(location);
+        for (long r = 0; r < vimg.nr(); ++r)
+        {
+            for (long c = 0; c < vimg.nc(); ++c)
+            {
+                interp(vchip, tf(dlib::vector<double, 2>(c, r)), vimg[r][c]);
```

Comment:
```suggestion
                interp(vchip, tf(dpoint(c, r)), vimg[r][c]);
```
There is a `dpoint` alias for this (i.e. double point)